users/tests/tests.py

```diff
@@ -122,67 +122,37 @@ def test_user_detail__logged_in_user_exists__renders_profile_page(self):
         self.assertIn(self.user.phone, content)
 
 
-# TODO test for changing info when not logged in and if changes are actually made
-class UserFormTest(TestCase):
-
-    @classmethod
-    def setUpTestData(cls):
-        cls.user = UserFactory()
-        cls.user.set_password('password')
-        cls.user.save()
-        cls.user_detail_url = reverse(user_detail, kwargs={'user_id': cls.user.id})
-
-    def test_UserForm_valid(self):
-        form = UserForm(data={
-            'first_name': "Alexander", 'last_name': "Orvik",
-            'phone': "45087749", 'study': "Samf", 'biography': "Nå er det endret", 'email': "alexaor@stud.ntnu.no"})
-        self.assertTrue(form.is_valid())
-
-    def test_UserForm_invalid(self):
-        form = UserForm(data={
-            'first_name': "", 'last_name': "",
-            'phone': "", 'study': "", 'email': ""})
-
-        self.assertFalse(form.is_valid())
-
-    def test_update_user_view_valid(self):
-        self.client.login(username=self.user.email, password='password')
-        response = self.client.post('/users/142/update', {
-            'first_name': "Alexander", 'last_name': "Orvik",
-            'phone': "45087749", 'study': "Samf", 'biography': "Nå er det endret", 'email': "alexaor@stud.ntnu.no"})
-        self.assertEqual(response.status_code, 302)
-
-    def test_update_user_get_request(self):
-        self.client.login(username=self.user.username, password='password')
-        response = self.client.get(reverse(update_user, kwargs={'user_id': self.user.id}))
-        self.assertTemplateUsed(response, 'users/update_profile_page.html')
-
-    def test_update_user_post_request_valid(self):
-        self.client.login(username=self.user.username, password='password')
-        response = self.client.post(reverse(update_user, kwargs={'user_id': self.user.id}), data={
-            'first_name': "Alexander", 'last_name': "Orvik",
-            'phone': "45087749", 'study': "Samf", 'biography': "Nå er det endret", 'email': "alexaor@stud.ntnu.no"})
+class UserUpdateViewTest(TestCase):
+    def setUp(self):
+        self.user = UserFactory.create()
+        self.client.force_login(self.user)
+        self.data = {
+            'first_name': 'Alexander',
+            'last_name': 'Orvik',
+            'phone': self.user.phone,
+            'study': self.user.study,
+            'biography': self.user.biography,
+            'email': self.user.email,
+            'home_address': self.user.home_address,
+            'study_address': self.user.study_address,
+            'in_relationship': self.user.in_relationship,
+            'ksg_status': self.user.ksg_status,
+            'commission': self.user.commission.id
+        }
```

Comment:
Tip: You can use `UserFactory.build()` and get an unsaved User object that you can pick attributes from 😉